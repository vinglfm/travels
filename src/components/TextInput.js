import React, { PureComponent } from 'react';
import styles from './Auth/AuthModal.css';

export default class TextInput extends PureComponent {
  render() {
    return (
      <div>
        <label className={styles.modal__form__label}>{this.props.label}{this.props.required ? <span className={styles.modal__form__label__required}>*</span> : ''}</label>
        <input className={!this.props.error || !this.props.value ? styles.modal__form__input : styles.modal__form__input__error} name={this.props.name} type={this.props.type} onChange={this.props.onChange} />
        <div className={styles.modal__form__error}>
          <span className={`${styles.modal__form__error__message} ${this.props.value && this.props.error ? '' : styles.modal__form__error__message__hidden}`}>{this.props.errorMessage}</span>
        </div>
      </div>
    );
  }
}