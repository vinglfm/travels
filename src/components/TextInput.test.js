import enzyme, {shallow} from 'enzyme';
import ReactAdapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new ReactAdapter() });

import React from 'react';
import TextInput from './TextInput';

describe('<TextInput />', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(<TextInput />);
  });

  it('renders text input pure component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
