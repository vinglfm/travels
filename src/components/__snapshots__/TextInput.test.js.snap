// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> renders text input pure component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          className={undefined}
>
          
</label>,
        <input
          className={undefined}
          name={undefined}
          onChange={undefined}
          type={undefined}
/>,
        <div
          className={undefined}
>
          <span
                    className="undefined undefined"
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            "",
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          undefined,
          "",
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": undefined,
          "name": undefined,
          "onChange": undefined,
          "type": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="undefined undefined"
/>,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "undefined undefined",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className={undefined}
>
            
</label>,
          <input
            className={undefined}
            name={undefined}
            onChange={undefined}
            type={undefined}
/>,
          <div
            className={undefined}
>
            <span
                        className="undefined undefined"
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              "",
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            undefined,
            "",
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": undefined,
            "name": undefined,
            "onChange": undefined,
            "type": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="undefined undefined"
/>,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "undefined undefined",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
